src/lib/configure_dynatrace.py

```diff
@@ -0,0 +1,136 @@
+import asyncio
+import re
+import os
+from os import listdir
+from os.path import isfile
+from typing import NamedTuple, List, Optional, Dict 
+
+from aiohttp import ClientSession
+
+from lib.context import LoggingContext
+
+from lib.fast_check import get_dynatrace_token_metadata
+from lib.credentials import fetch_dynatrace_url, fetch_dynatrace_api_key
+import yaml
+import json
+from main import is_yaml_file
+
+
+class ConfigureDynatrace:
+    async def post_dashboard(self, dynatrace_url: str, dynatrace_api_key: str, path: str, name:str, timeout: Optional[int] = 2) -> List[Dict]:
+        try:
+            with open(path, encoding="utf-8") as dashboard_file:
+                 dashboard_json = json.load(dashboard_file)             
+            response = await self.session.post(              
+                url=f"{dynatrace_url.rstrip('/')}/api/config/v1/dashboards",
+                headers={
+                    "Authorization": f"Api-Token {dynatrace_api_key}",
+                    "Content-Type": "application/json; charset=utf-8"
+                },
+                timeout=timeout,
+                json=dashboard_json)
+            if response.status != 201:
+                response_json=  await response.json()                
+                if 'owner' in json.dumps(response_json):                                        
+                    del dashboard_json["dashboardMetadata"]["owner"]
+                    response = await self.session.post(              

```

Comment:
To make this more readable you could add `remove_owner: bool = False` argument to `post_dashboard` function. Function could check this argument to determine whether to perform ` del dashboard_json["dashboardMetadata"]["owner"]` before sending request, and to check if it should retry without `owner` field.